### PR TITLE
[refactor](nereids) Refine some code snippets

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

~~Issue Number: close #xxx~~

## Problem Summary:

Refine some code snippets:
1. Rename: ExpressionUtils::add -> ExpressionUtils::and
2. Reduce temporary objects when combing expressions.

## Checklist(Required)

1. Does it affect the original behavior: No
3. Has unit tests been added: No Need
4. Has document been added or modified: No Need
5. Does it need to update dependencies: No
6. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java [hunk 1]
```diff
@@ -89,51 +87,36 @@ public static Expression or(List<Expression> expressions) {
...
-            if (op == NodeType.AND) {
-                return new Literal(true);
-            }
-            if (op == NodeType.OR) {
-                return new Literal(false);
-            }
-        }
-
-        if (expressions.size() == 1) {
+            return new Literal(op == NodeType.AND);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java:93]: If u do that, u need add a check at the top of this function to check `NodeType` is either `AND` or `OR`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/Utils.java [hunk 2]
```diff
@@ -28,10 +28,7 @@ public class Utils {
      * @return quoted string
      */
     public static String quoteIfNeeded(String part) {
-        if (part.matches("[a-zA-Z0-9_]+") && !part.matches("\\d+")) {
-            return part;
-        } else {
-            return part.replace("`", "``");
-        }
+        return part.matches("\\w*[\\w&&[^\\d]]+\\w*")
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/Utils.java:31]: this pattern is not intuitional.  it is better add a comment to explain the pattern means all legal string except pure digit string.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java [hunk 3]
```diff
@@ -89,51 +87,36 @@ public static Expression or(List<Expression> expressions) {
...
+                distinctExpressions.add(expression);
             }
-            distinctExpressions = distinctExpressions.stream().filter(p -> !p.equals(Literal.TRUE_LITERAL))
-                    .collect(Collectors.toList());
         }
 
-        if (op == NodeType.OR) {
-            if (distinctExpressions.contains(Literal.TRUE_LITERAL)) {
-                return Literal.TRUE_LITERAL;
+        List<Expression> result = Lists.newArrayListWithCapacity(distinctExpressions.size() / 2 + 1);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java:109]: maybe we could use stream reduce api to do this without recursion

Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java:111]: The output would not be the same as the original one if we used `stream reduce` api.

recursion: `(A AND B) AND (C AND D)`
reduce: `((A AND B) AND C) AND D)`

If the inconsistency is acceptable, I will use the `reduce` way to simplify it.

Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java]: Simplified it by `reduce` api.

**Review by morrySnow — APPROVED** [review]

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**